api/server/v1alpha1/device_types.go

```diff
@@ -46,6 +46,7 @@ type DeviceSpec struct {
 	ProjectID    string   `json:"projectID"`
 	UserData     string   `json:"userdata,omitempty"`
 	Tags         []string `json:"tags,omitempty"`
+	Locked       *bool    `json:"locked,omitemtpy"`
```

Comment:
In looking through `packngo` it appears that `Locked` is present in [`DeviceUpdateRequest`](https://github.com/packethost/packngo/blob/6292a62434aeffca0d2cca5c2079349c2fc64650/devices.go#L166) but not `DeviceCreateRequest`. From what I can tell, you cannot create a device that is locked, but you can lock a created device. So if a user specifies `locked: true` in their spec on create here, then it would just get locked on the first call to `Update()` after the device is provisioned. Is my logic correct here?

Reply:
Yes you are correct. Create request does not accept `Locked` parameter.